motionplan/motionPlanner.go

```diff
@@ -433,178 +432,42 @@ IK:
-		if err != nil {
-			return nil, err
-		}
-		nextInputSlice, err := sf.mapToSlice(nextInput)
-		if err != nil {
-			return nil, err
-		}
-		// If we are working with a PTG plan we redefine the startConfiguration in terms of the endConfiguration.
-		// This allows us the properly interpolate along the same arc family and sub-arc within that family.
-		if relative {
-			currInputSlice = nextInputSlice
-		}
-		return &ik.Segment{
-			StartPosition:      currPose,
-			EndPosition:        nextPose,
-			StartConfiguration: currInputSlice,
-			EndConfiguration:   nextInputSlice,
-			Frame:              sf,
-		}, nil
-	}
-
-	// iterate through remaining plan and append remaining segments to check
-	for i := wayPointIdx; i < len(offsetPlan.Path())-1; i++ {
-		segment, err := createSegment(poses[i], poses[i+1], offsetPlan.Trajectory()[i], offsetPlan.Trajectory()[i+1])
-		if err != nil {
-			return err
-		}
-		segments = append(segments, segment)
-	}
-
-	// go through segments and check that we satisfy constraints
-	// TODO(RSDK-5007): If we can make interpolate a method on Frame the need to write this out will be lessened and we should be
-	// able to call CheckStateConstraintsAcrossSegment directly.
-	var totalTravelDistanceMM float64
-	for _, segment := range segments {
-		interpolatedConfigurations, err := interpolateSegment(segment, sfPlanner.planOpts.Resolution)
-		if err != nil {
-			return err
-		}
-		for _, interpConfig := range interpolatedConfigurations {
-			poseInPath, err := sf.Transform(interpConfig)
-			if err != nil {
-				return err
-			}
-
-			// Check if look ahead distance has been reached
-			currentTravelDistanceMM := totalTravelDistanceMM + poseInPath.Point().Distance(segment.StartPosition.Point())
-			if currentTravelDistanceMM > lookAheadDistanceMM {
-				return nil
-			}
-
-			// If we are working with a PTG plan the returned value for poseInPath will only
-			// tell us how far along the arc we have traveled. Since this is only the relative position,
-			//  i.e. relative to where the robot started executing the arc,
-			// we must compose poseInPath with segment.StartPosition to get the absolute position.
-			interpolatedState := &ik.State{Frame: sf}
-			if relative {
-				interpolatedState.Position = spatialmath.Compose(segment.StartPosition, poseInPath)
-			} else {
-				interpolatedState.Configuration = interpConfig
-			}
-
-			// Checks for collision along the interpolated route and returns a the first interpolated pose where a collision is detected.
-			if isValid, err := sfPlanner.planOpts.CheckStateConstraints(interpolatedState); !isValid {
-				return fmt.Errorf("found error between positions %v and %v: %s",
-					segment.StartPosition.Point(),
-					segment.EndPosition.Point(),
-					err,
-				)
-			}
-		}
-
-		// Update total traveled distance after segment has been checked
-		totalTravelDistanceMM += segment.EndPosition.Point().Distance(segment.StartPosition.Point())
+	if sfPlanner.useTPspace {
```

Comment:
I like that we have two separate functions for this now. 
Initially, I was against it since we might be incurring tech-debt with maintaining this, but I think this ok for now.
I also wonder if we want to commit to having this bifurcation of how we check plans stay with us in the future or if we want to build such that we can once again have only one function to check plans, regardless of their relativity.